pki/pki.go

```diff
@@ -121,12 +129,22 @@ type StatusInfo struct {
 	FailInfo     asn1.BitString `asn1:"optional"`
 }
 
-// ParseFailInfo parses the FailInfo field of a PKIStatusInfo to PKIFailureInfo
-func (psi StatusInfo) ParseFailInfo() (FailureInfo, error) {
-	for _, pfi := range FailureInfos {
-		if psi.FailInfo.At(int(pfi)) != 0 {
-			return pfi, nil
+// Err return nil when si Status is StatusGranted or StatusGrantedWithMods
+//
+// Otherwise, Err returns an error with FailInfo if any.
+func (si StatusInfo) Err() error {
+	var err error
+	if si.Status != StatusGranted && si.Status != StatusGrantedWithMods {
+		for _, fi := range failureInfos {
+			if si.FailInfo.At(int(fi)) != 0 {
+				err = errors.Join(err, fi.Error())
```

Comment:
Instead of joining one by one, can you create an array then join together?

It is better to read the original source code: https://cs.opensource.google/go/go/+/refs/tags/go1.22.4:src/errors/join.go;l=19

Reply:
Updated.